pkgs/lib.nix

```diff
@@ -15,6 +15,11 @@ let self = {
       MemoryDenyWriteExecute = "true";
       ProtectKernelTunables = "true";
       ProtectKernelModules = "true";
+      ProtectKernelLogs = "true";
+      ProtectClock = "true";
+      # Enable these when systemd v247 is available
+      # ProtectProc = "invisible";
+      # ProcSubset = "pid";
```

Comment:
We may not actually want to set this because the systemd.exec manpage says for `ProcSubset`
> Note that Linux exposes various kernel APIs via /proc/, which are made unavailable with this setting. Since these APIs are used frequently this option is useful only in a few, specific cases, and is not suitable for most non-trivial programs.

Reply:
Fixed in 477684c

Changed wording to specify that we need to carefully test these before we enable them